test/gtest/uct/ib/test_ud.cc

```diff
@@ -918,6 +918,90 @@ UCS_TEST_SKIP_COND_P(test_ud, ctls_loss,
 
 UCT_INSTANTIATE_UD_TEST_CASE(test_ud)
 
+
+class test_ud_peer_failure : public ud_base_test {
+public:
+    test_ud_peer_failure()
+    {
+        m_err_count = 0;
+    }
+
+    void init()
+    {
+        set_config("UD_TIMEOUT?=3s");
+        ud_base_test::init();
+    }
+
+    void kill_receiver()
+    {
+        m_entities.remove(m_e2);
+    }
+
+    static ucs_status_t err_cb(void *arg, uct_ep_h ep, ucs_status_t status)
+    {
+        m_err_count++;
+        return UCS_OK;
+    }
+
+    uct_error_handler_t get_err_handler() const
+    {
+        return err_cb;
+    }
+
+protected:
+    static size_t m_err_count;
+};
+
+size_t test_ud_peer_failure::m_err_count = 0;
+
+UCS_TEST_SKIP_COND_P(test_ud_peer_failure, send_am_after_kill,
+                     !check_caps(UCT_IFACE_FLAG_AM_SHORT |
+                                 UCT_IFACE_FLAG_EP_CHECK),
+                     // Increase UD timer tick to not get NO_RESOURCE from AM
+                     "UD_TIMER_TICK?=2s") {
+    ucs_status_t status;
+
+    connect();
+
+    scoped_log_handler slh(wrap_errors_logger);
```

Comment:
why needed?
i'd expect adding the error handler to suppress error messages

Reply:
removed